### PR TITLE
Add and export GetHostTags function for python checks

### PR DESCRIPTION
**Pull request by natashadada**
### What does this PR do?

This PR adds a GetHostTags and creates a c binding so this can be used from python integrations.

### Motivation

Metrics emitted by DBM integrations (postgres, sql server, and mysql) are tagged with the tags from the database instance host. We would like to introduce a user-facing setting (default false) that allows us to also consider the tags of the agent host. If this setting is enabled, we will call this function from the python integrations to get the agent host tags and use these to tag metrics.

### Describe how to test/QA your changes

Three options for testing:

#### Option 1
I followed [these steps](https://datadoghq.atlassian.net/wiki/spaces/~519636516/pages/3968532776/Test+a+change+across+datadog-agent+and+integrations-core) to test:
1. Created an ec2 instance
2. Installed this version of the agent
3. Called this function from the postgres integration in the integrations-core repo and logging the result.

#### Option 2 (easiest)

Test this with the postgres check by calling `get_agent_host_tags` and logging the result [here](https://github.com/DataDog/integrations-core/blob/4ca816369870b9bd2d3f094fc614c58c2a065ca8/postgres/datadog_checks/postgres/postgres.py#L1014). 

#### Option 3

Create a custom check in checks.d:

test_check.py:
```
from datadog_checks.base import AgentCheck
from datadog_checks.base.utils.db.utils import get_agent_host_tags

class HelloCheck(AgentCheck):
    def check(self, instance):
        data = dict()
        data['agent_host_tags'] = get_agent_host_tags()
        data['message'] = 'testing get agent host tags function'
        self.send_log(data)
```

And add a configuration for in conf.d:

test_check.yaml:
```
init_config:
instances:
  [{}]

logs:
  - type: integration
    source: my_source
    service: my_service
```

You can add tags to an instance and verify they appear in logs.

### CODE REVIEW

**Review by lu-zhengda — APPROVED** [review]

File: releasenotes/notes/get-host-tags-in-python-checks-3d6c7d9b670eb5e5.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Expose agent's get host tags function to python checks via new `datadog_agent.get_host_tags` method
```
Comment (maycmlee) [inline releasenotes/notes/get-host-tags-in-python-checks-3d6c7d9b670eb5e5.yaml:11]:
```suggestion
    Expose the Agent's get host tags function to python checks using the new `datadog_agent.get_host_tags` method.
```

**Review by maycmlee — APPROVED** [review]
Left a couple of small suggestions, but approving.

File: pkg/collector/python/datadog_agent.go [hunk 2]
```diff
@@ -58,6 +59,18 @@ func GetHostname(hostname **C.char) {
 	*hostname = TrackedCString(goHostname)
 }
 
+// GetHostTags exposes the tags of the agent host to Python checks.
+//
+//export GetHostTags
+func GetHostTags(hostTags **C.char) {
+	tags := hosttags.Get(context.Background(), true, config.Datadog())
```
Comment (jmeunier28) [inline pkg/collector/python/datadog_agent.go:66]: does it make sense to set a timeout here? is it possible collecting these tags can take a while? we probably don't want the agent integration init to just hang forever waiting for this to return (if that's possible)

Comment (natashadada) [reply pkg/collector/python/datadog_agent.go:66]: good point! I think it's safest to add a timeout. I noticed the [Get](https://github.com/DataDog/datadog-agent/blob/f370d436b6f6e3d80e8868f68c7db2f6d89ad0cf/comp/metadata/host/hostimpl/hosttags/tags.go#L102) function that we're calling doesn't return an error so we wouldn't have a way of knowing if the context timed out.

There are a few options I can think of to solve that:
1. Update that function to return an error and update the places that call it 
2. I could rename this function `MustGet` and add a `Get` function that does return an error and call that one
> is it possible collecting these tags can take a while?
3. I don't know the answer to this. I'll investigate today and if the answer ends up being no then maybe we don't need a timeout at all.

I'm going to look into how long it takes to collect tags regardless, but right now I'm leaning toward option 2. What do you think?

Comment (gh123man) [reply pkg/collector/python/datadog_agent.go:66]: Just to add my two cents here. 

Tags are fetched at startup and the cache is populated. As far as I understand (from reading the code) [the cache is never expired, bypassed or invalidated. ](https://github.com/DataDog/datadog-agent/blob/ca8c80b7c199ef543e14a907a10b0cd7c0e34dee/comp/metadata/host/hostimpl/hosttags/tags.go#L103)

So as long as: 
1. you pass `cache: true` to `Get` (which you are), 
2. you are not calling `Get` for the very first time

this function should always return immediately and never block.

**Review by gh123man — APPROVED** [review]

File: pkg/collector/python/datadog_agent.go (hunk 2 above)
Comment (jmeunier28) [reply pkg/collector/python/datadog_agent.go:66]: oh nice, i didn't notice this [code block](https://github.com/DataDog/datadog-agent/blob/ca8c80b7c199ef543e14a907a10b0cd7c0e34dee/comp/metadata/host/hostimpl/hosttags/tags.go#L103-L108) when i looked at the function originally. That makes sense, so need to change anything @natashadada 👍 

I am still curious about this though:
> also, should we be worried about non utf-8 compliant data? should we sanitize the tag list to make sure that json marshaling won't fail bc of a bad tag?

if a tag set somehow contains bad values we will fail to return any tags. Maybe this is fine and customers can debug based on logs?

Comment (jmeunier28) [reply pkg/collector/python/datadog_agent.go:66]: > I'm also not sure there's a good way to go through each tag and identify bad values.

you could check that each tag value is valid UTF-8. I'm not sure if this happens other places in the pipeline. 

failing for all tags && just doing this log line seems fine to me though:

> log.Warnf("Error getting host tags: %v. Invalid tags: %v", err, tags)

Comment (natashadada) [reply pkg/collector/python/datadog_agent.go:66]: ok yeah I agree, I'll update the error message and that should be good (and we can always adjust later if we need to). Thanks for looking through all of this 🙏

**Pull request merged**